src/core/app.lua

```diff
@@ -51,10 +51,13 @@ local function with_restart (app, method)
 end
 
 -- Restart dead apps.
+local actions = { start={}, restart={}, reconfig={}, keep={}, stop={} }
 function restart_dead_apps ()
+   if not use_restart then return end
    local restart_delay = 2 -- seconds
-   local actions = { start={}, restart={}, reconfig={}, keep={}, stop={} }
    local restart = false
+   actions.restart[1] = nil
```

Comment:
I would favour the traversal method, but maybe we shouldn't bother at all as long as that logic is bypassed when `use_restart` is `false`. The dead-app detection code has other performance issues that need to be addressed at some point.